src/Microsoft.ML.TimeSeries/SsaAnomalyDetectionBase.cs

```diff
@@ -82,11 +82,29 @@ public static Double SquaredDifference(Double actual, Double predicted)
     }
 
     /// <summary>
-    /// This base class that implements the general anomaly detection transform based on Singular Spectrum modeling of the time-series.
+    /// The wrapper to the base class that implements the general anomaly detection transform based on Singular Spectrum modeling of the time-series.
     /// For the details of the Singular Spectrum Analysis (SSA), refer to http://arxiv.org/pdf/1206.6910.pdf.
     /// </summary>
-    public abstract class SsaAnomalyDetectionBase : SequentialAnomalyDetectionTransformBase<Single, SsaAnomalyDetectionBase.State>
+    public class SsaAnomalyDetectionBaseWrapper : IStatefulTransformer, ICanSaveModel
     {
+        public bool IsRowToRowMapper => Base.IsRowToRowMapper;
+
+        IStatefulTransformer IStatefulTransformer.Clone() => Base.Clone();
+
+        public Schema GetOutputSchema(Schema inputSchema) => Base.GetOutputSchema(inputSchema);
+
+        public IRowToRowMapper GetRowToRowMapper(Schema inputSchema) => Base.GetRowToRowMapper(inputSchema);
+
+        public IRowToRowMapper GetStatefulRowToRowMapper(Schema inputSchema) => ((IStatefulTransformer)Base).GetStatefulRowToRowMapper(inputSchema);
+
+        public IDataView Transform(IDataView input) => Base.Transform(input);
+
+        public virtual void Save(ModelSaveContext ctx) => Base.SaveThis(ctx);
+
+        internal IStatefulRowMapper MakeRowMapper(Schema schema) => Base.MakeRowMapper(schema);
+
+        internal IDataTransform MakeDataTransform(IDataView input) => Base.MakeDataTransform(input);
+
         public abstract class SsaArguments : ArgumentsBase
```

Comment:
>SsaArguments [](start = 30, length = 12)

Can you rename them to `SsaOptions`? and also rename local variables to `options`
If  the arguments are not used in other places can we make them internal?